datafusion/functions-nested/src/reverse.rs

```diff
@@ -183,6 +195,75 @@ fn general_array_reverse<O: OffsetSizeTrait + TryFrom<i64>>(
     )?))
 }
 
+fn list_view_reverse<O: OffsetSizeTrait + TryFrom<i64>>(
+    array: &GenericListViewArray<O>,
+    field: &FieldRef,
+) -> Result<ArrayRef> {
+    let (_, offsets, sizes, values, nulls) = array.clone().into_parts();
+
+    // Construct indices, sizes and offsets for the reversed array by iterating over
+    // the list view array in the logical order, and reversing the order of the elements.
+    // We end up with a list view array where the elements are in order,
+    // even if the original array had elements out of order.
+    let mut indices: Vec<O> = Vec::with_capacity(values.len());
+    let mut new_sizes = Vec::with_capacity(sizes.len());
+    let mut new_offsets: Vec<O> = Vec::with_capacity(offsets.len());
+    let mut new_nulls =
+        Vec::with_capacity(nulls.clone().map(|nulls| nulls.len()).unwrap_or(0));
+    new_offsets.push(O::zero());
+    let has_nulls = nulls.is_some();
+    for (i, offset) in offsets.iter().enumerate().take(offsets.len()) {
+        // If this array is null, we set the new array to null with size 0 and continue
+        if let Some(ref nulls) = nulls {
+            if nulls.is_null(i) {
+                new_nulls.push(false); // null
+                new_sizes.push(O::zero());
+                new_offsets.push(new_offsets[i]);
+                continue;
+            } else {
+                new_nulls.push(true); // valid
+            }
+        }
+
+        // Each array is located at [offset, offset + size), so we collect indices in the reverse order
+        let array_start = offset.as_usize();
+        let array_end = array_start + sizes[i].as_usize();
+        for idx in (array_start..array_end).rev() {
+            indices.push(O::usize_as(idx));
+        }
```

Comment:
I feel this roundtripping from O -> usize -> O could be simplified, perhaps if we use a loop instead of a for? Thoughts? 🤔

Reply:
Don't catch what you mean by "a loop instead of a for" though! Mind explaining?